File: app/src/main/java/com/indisparte/pothole/data/model/Pothole.java
package com.indisparte.pothole.data.model;

import androidx.annotation.NonNull;

import java.util.Objects;

/**
 * @author Antonio Di Nuzzo (Indisparte)
 */
public class Pothole {
    private final String user;
    private final Double lat, lon, var;

    public Pothole(@NonNull String user,
                   @NonNull Double lat,
                   @NonNull Double lon,
                   @NonNull Double var
    ) {
        this.user = user;
        this.lat = lat;
        this.lon = lon;
        this.var = var;
    }

    public Double getLat() {
        return lat;
    }

    public Double getLon() {
        return lon;
    }

    public Double getVar() {
        return var;
    }

    public String getUser() {
        return user;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Pothole)) return false;
        Pothole pothole = (Pothole) o;
        return getUser().equals(pothole.getUser()) && getLat().equals(pothole.getLat()) && getLon().equals(pothole.getLon()) && getVar().equals(pothole.getVar());
    }

    @Override
    public int hashCode() {
        return Objects.hash(getUser(), getLat(), getLon(), getVar());
    }

    @NonNull
    @Override
    public String toString() {
        return "Pothole{" +
                "user='" + user + '\'' +
                ", latitude=" + lat +
                ", longitude=" + lon +
                ", variation=" + var +
                '}';
    }
}

File: app/src/main/java/com/indisparte/pothole/data/model/Pothole.kt
package com.indisparte.pothole.data.model

/**
 * @author Antonio Di Nuzzo (Indisparte)
 */
data class Pothole(
    val user: String,
    val lat: Double,
    val lon: Double,
    val variation: Double
) {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is Pothole) return false

        if (user != other.user) return false
        if (lat != other.lat) return false
        if (lon != other.lon) return false
        if (variation != other.variation) return false

        return true
    }

    override fun hashCode(): Int {
        var result = user.hashCode()
        result = 31 * result + lat.hashCode()
        result = 31 * result + lon.hashCode()
        result = 31 * result + variation.hashCode()
        return result
    }
}